sw/device/lib/dif/dif_pwrmgr.h

```diff
@@ -0,0 +1,499 @@
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_req_type_t req_type, uint8_t source,
+    bool *request);
+
+/**
+ * Enable or disable wake-up requests recording.
+ *
+ * The power manager automatically starts recording wake-up requests when it
+ * begins a valid low power entry. Recording continues until it is explicitly
+ * disabled by calling this function.
+ *
+ * @param pwrmgr A power manager.
+ * @param new_state Whether power manager should record wake-up requests.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_wakeup_request_recording_enabled_set(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_toggle_t new_state);
+
+/**
+ * Check if wake-up requests are being recorded.
+ *
+ * @param pwrmgr A power manager.
+ * @param cur_state Whether the power manager is recording wake-up requests.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_wakeup_request_recording_enabled_get(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_toggle_t *cur_state);
+
+/**
+ * Wake-up types.
+ */
+typedef enum dif_pwrmgr_wakeup_type {
+  /**
+   * Wake-up due to a peripheral source request.
+   */
+  kDifPwrmgrWakeupTypeSource,
+  /**
+   * Despite low power mode being enabled and executing a wait for interrupt
+   * (WFI) instruction, an interrupt arrived at just the right time to break the
+   * executing core out of WFI.
+   */
+  kDifPwrmgrWakeupTypeFallThrough,
+  /**
+   * Despite low power mode being enabled and executing a wait for interrupt
+   * (WFI) instruction, an active flash, life cycle, or OTP operation was
+   * in progress when the power controller attempted to initiate low power
+   * entry.
+   */
+  kDifPwrmgrWakeupTypeAbort,
+} dif_pwrmgr_wakeup_type_t;
+
+/**
+ * Wake-up reason and source requests since the last time recording started.
+ */
+typedef struct dif_pwrmgr_wakeup_reason {
+  /**
+   * Wake-up type.
+   */
+  dif_pwrmgr_wakeup_type_t wakeup_type;
+  /**
+   * Peripheral sources that requested wake-up since the last time recording has
+   * started.
+   */
+  uint32_t sources;
+} dif_pwrmgr_wakeup_reason_t;
+
+/**
+ * Get wake-up reason and source requests since the last time recording started.
+ *
+ * The power manager automatically starts recording wake-up requests when it
+ * begins a valid low power entry. Recording continues until it is explicitly
+ * disabled by calling `dif_pwrmgr_wakeup_request_recording_enabled_set`. Thus,
+ * it is possible to record wake-up requests for multiple peripheral sources.
+ *
+ * @param pwrmgr A power manager.
+ * @param cur_config Whether power manager is recording wake-up reasons.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_wakeup_reason_get(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_wakeup_reason_t *reason);
+
+/**
+ * Clear wake-up reason and requests recorded during last low power state.
```

Comment:
I remember in the past reviews @mcy felt strongly about function descriptions having form of  "Clears..." instead of "Clear...".